addons/imaginary.sh

```diff
@@ -0,0 +1,144 @@
+lowest_compatible_nc 24
+
+# Only applies if previewgenerator is installed
+if ! is_app_installed previewgenerator
+then
+    msg_box "Imaginary is only needed if your use the Preview Generator app, no need to run this script."
+    exit
+fi
+
+# Check if Imaginary is already installed
+if ! does_this_docker_exist nextcloud/aio-imaginary
+then
+    # Ask for installing
+    install_popup "$SCRIPT_NAME"
+else
+    # Ask for removal or reinstallation
+    reinstall_remove_menu "$SCRIPT_NAME"
+    # Removal
+    if yesno_box_no "Do you want to remove the Imaginary docker container and settings?"
+    then
+        docker-compose_down "$SCRIPTS"/imaginary-docker/docker-compose.yml
+        nextcloud_occ config:system:delete enabledPreviewProviders
+        nextcloud_occ config:system:delete preview_imaginary_url
+        rm -rf "$SCRIPTS"/imaginary-docker
+    fi
+    # Show successful uninstall if applicable
+    removal_popup "$SCRIPT_NAME"
+fi
+
+# Generate docker-compose.yml
+mkdir -p "$SCRIPTS"/imaginary-docker
+if [ ! -f "$SCRIPTS/imaginary-docker/docker-compose.yml" ]
+then
+    touch "$SCRIPTS/imaginary-docker/docker-compose.yml"
+    cat << IMAGINARY_DOCKER_CREATE > "$SCRIPTS"/imaginary-docker/docker-compose.yml
+version: '3.1'
+services:
+  imaginary:
+    image: nextcloud/aio-imaginary:latest
+    container_name: imaginary
+    restart: always
+    environment:
+       PORT: 9000
+    command: -concurrency 50 -enable-url-source -log-level debug
+    ports:
+      -  127.0.0.1:9000:9000
+IMAGINARY_DOCKER_CREATE
+    print_text_in_color "$IGreen" "SCRIPTS/imaginary-docker/docker-compose.yml was successfully created."
+fi
+
+# Start the container
+docker compose -p imaginary -f "$SCRIPTS"/imaginary-docker/docker-compose.yml up -d
+
+# Test if imaginary is working
+countdown "Testing if it works in 3 sedonds" "3"
+if curl -O "http://127.0.0.1:9000/crop?width=500&height=400&url=https://raw.githubusercontent.com/h2non/imaginary/master/testdata/large.jpg"
+then
+    print_text_in_color "$IGreen" "imaginary seems to be working OK!"
+else
+    msg_box "Test failed, please report this to: $ISSUES"
+    exit
+fi
+
+# Install preview generator just in case
+install_and_enable_app previewgenerator
+
+# check if the previewgenerator is installed and enabled
+if is_app_enabled previewgenerator
+then
+    # enable previews
+    nextcloud_occ config:system:set enable_previews --value=true --type=boolean
+
+    # install needed dependency for movies
+    install_if_not ffmpeg
+else
+    exit
+fi
+
+# Set providers
+nextcloud_occ config:system:set enabledPreviewProviders 0 --value="OC\\Preview\\JPEG"
+nextcloud_occ config:system:set enabledPreviewProviders 1 --value="OC\\Preview\\PNG"
+nextcloud_occ config:system:set enabledPreviewProviders 2 --value="OC\\Preview\\WEBP"
+nextcloud_occ config:system:set enabledPreviewProviders 3 --value="OC\\Preview\\HEIF"
+nextcloud_occ config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\TIFF"
```

Comment:
You dont need to add this as it is enabled automatically by enabling the imaginary preciew provider. Same with others.

Reply:
But it's needed for the "rebuild", right?